.github/workflows/test.yml

```
@@ -0,0 +1,15 @@
+name: test
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+on:
+  pull_request: { types: [opened, reopened, synchronize, ready_for_review] }
+  push: { branches: [ main ] }
+
```

```suggestion
permissions:
  contents: read
```
Any time I turn on code scanning for GHA, I get dinged for not doing this, and it is good practice, so 🤷‍♀️.